framework/decode/vulkan_feature_util.cpp

```diff
@@ -129,6 +129,25 @@ bool IsSupportedExtension(const std::vector<VkExtensionProperties>& properties,
     return false;
 }
 
+// There are some extensions which can be enabled by the application, but can be ignored during replay if
+// they don't exist.
+// For example, the VK_EXT_tooling_info extension is enabled by the GfxReconstruct capture layer
+// and may be queried/enabled during instance creation, but is not used for anything other than
+// querying layer information.  This can be problematic if the instance replaying attempts to
+// enable the extension with no support present on the replay device (usually because the layer is
+// no longer there)
+bool IsIgnorableExtension(const char* extension)
+{
+    assert(extension != nullptr);
+
+    if (!strcmp(VK_EXT_TOOLING_INFO_EXTENSION_NAME, extension))
```

Comment:
We're doing the loop on line 180 below.  This just organizes the extension checks in this one place.  Similar to the existing `IsSupportedExtension`.

Reply:
However, I do see there is a `util::platform::StringCompare` function I should use instead of `strcmp`.  Let me know if you still want me to make the other change.

Reply:
The loop below is looping through the list of extensions passed in.

My point is that if we have to add any more functions to the list of ignorable functions.  you'll have to alter this function, and I think it's cleaner for it to be in an initialized map.  https://github.com/LunarG/gfxreconstruct/blob/d6dcd2baea4f98d4572cda3217197c4aa2505372/layer/trace_layer.cpp#L58 is an example of this where we keep adding extensions.  (But trace_layer.cpp doesn't quite go far enough because its list isn't in an `unordered_map`.)

Reply:
Do you mean `vector` instead of `unordered_map`?  Also, having a vector,list or anything of const char* will compare the pointers.  That fails.  I can do the following:

```
void RemoveIgnorableExtensions(const std::vector<VkExtensionProperties>& properties,
                               std::vector<const char*>*                 extensions)
{
    assert(extensions != nullptr);

    std::vector<std::string> ignorable_extensions;
    ignorable_extensions.push_back(VK_EXT_TOOLING_INFO_EXTENSION_NAME);

    auto extensionIter = extensions->begin();
    while (extensionIter != extensions->end())
    {
        if (!IsSupportedExtension(properties, *extensionIter))
        {
            if (std::find(ignorable_extensions.begin(), ignorable_extensions.end(), *extensionIter) !=
                ignorable_extensions.end())
            {
```

which completely removes this function if that's prefered.  But I worry that we're introducing more looping and complexity.  Although it is all pre-create Device.